operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/TemporaryResourceCache.java

```diff
@@ -60,7 +63,23 @@ protected boolean removeEldestEntry(java.util.Map.Entry<String, Boolean> eldest)
     }
   }
 
-  public synchronized void onEvent(T resource, boolean unknownState) {
+  public synchronized void onDeleteEvent(T resource, boolean unknownState) {
+    tombstones.put(resource.getMetadata().getUid(), System.currentTimeMillis());
+    onEvent(resource, unknownState);
+  }
+
+  public synchronized void onAddOrUpdateEvent(T resource) {
+    onEvent(resource, false);
+  }
+
+  synchronized void onEvent(T resource, boolean unknownState) {
+    if (!tombstones.isEmpty()) {
+      var iter = tombstones.entrySet().iterator();
+      var entry = iter.next();
```

Comment:
Not sure if understand this part, it always removes one random event from the tombstone if expired? Didn't you mean to iterate over all events?

Reply:
No, this is to amortorize the expiration logic - you could of course decide on a greater number than 1.

Reply:
Maybe desribe in there as a comment too, that is intentional, since I will at least naivly stop on this every time I open it :)

Reply:
I would like to see this as a comment, indeed, as well.